### DISCUSSION TITLE
how to print drift query for debugging.

### DISCUSSION THREAD

**Discussion by aneebali** [simolus3/drift#2516] [General]
Is anyone knows how to print drift query for debugging.

**Comment by simolus3** [comment]
You can use the `logStatements: true` parameter available on all constructors of a [NativeDatabase](https://drift.simonbinder.eu/api/drift.ffi/nativedatabase/createinbackground) to log statements - drift will print the SQL and associated variables in that case.

**Reply by Silko5** [reply]
Does this setting log SQL into some files or just prints them to console? Because in console queries gets cut as strings are too long. (more about the issue here: https://github.com/flutter/flutter/issues/22665)

**Reply by simolus3** [reply]
> Because in console queries gets cut as strings are too long.

By default, drift simply uses `print`. You can set [`driftRuntimeOptions.debugPrint`](https://drift.simonbinder.eu/api/drift/driftruntimeoptions/debugprint) to a function accepting the string to log. That way, you can set it to Flutter's `debugPrint` or another function potentially logging the SQL to a file.
